test/sanity/yamllint/yamllinter.py

```diff
@@ -45,8 +45,11 @@ def check(self, paths):
         for path in paths:
             extension = os.path.splitext(path)[1]
 
-            with open(path) as f:
-                contents = f.read()
+            with open(path, 'rb') as f:
+                if sys.version_info[0] == 3:
+                    contents = f.read().decode('utf-8')
+                else:
+                    contents = f.read()
```

Comment:
Couple notes:
* Instead of doing a version check here, we could just use to_native().  Your code will work but I don't necessarily trust that the next person to edit this will understand what's going on with the python version check and decode.
* For controller code we make everything text or explicitly leave it as bytes with a named variable.  I believe that @mattclay wants the same convention for ansible-test code.  This is kind of a plugin to ansible-test but I think we should probably follow the same convention here?

Reply:
From slack:

> mattclay: sivel abadger1999 We've avoided a lot of unicode issues in ansible-test by running with LC_ALL set, so most of the the encoding work in ansible-test has been on an as needed basis. Working towards removing the use of LC_ALL we'll want to go with the unicode sandwich, but we don't yet have good functions in place like _text provides to do that.

So we do want to go unicode sandwich here which means contents should become text on both python2 and python3.  Then other code later on will have to be changed to be sure that it's not mixing text and bytes.